tests/test_definitions/multicolumn_map_expectations/expect_compound_columns_to_be_unique.json

```diff
@@ -213,7 +213,7 @@
     },{
       "title": "unexpected_values_exact_match_out_without_unexpected_index_list",
       "exact_match_out" : true,
-      "suppress_test_for": ["pandas"],
+      "only_for": ["sqlalchemy"],
```

Comment:
@NathanFarmer I feel that we must make it (or an equivalent, additional test work for `Spark`).  The existing test excludes `Pandas`, because `unexpected_index_list` only applies to `Pandas`; however, the functionality *must* work for the other execution engines.  Since you introduced the "force-sorting" of both sides in the assertion in `self_check/util.py`, why would the existing test not work?  To me, having this test is critical.  Thank you!

Reply:
In [self_check/util.py#L1882](https://github.com/great-expectations/great_expectations/blob/82b3e29cf89792c8e7cb9bfc95b353c1498fa1fa/great_expectations/self_check/util.py#L1882) when `test["exact_match_out"] is True` all of the test sort logic is foregone. There is no analog to this test that would work that I can think of.

Reply:
Note: There is a test above this one for `pandas`. It is possible if we sort the data in the test harness it will work for `spark` but I don't think even that is guaranteed.

Reply:
@NathanFarmer I feel that we should discuss this in order to find a solution for testing the functionality for the `Spark` engine.  The reasoning is that the addition of `row_number()` and sorting on it was key to making Spark work.  However, I believe that you are pointing out that the sort order of the results is the only difference, the output being the same as expected otherwise.  If this is the case, then we need to figure out how to show that in a test.  If I am not understanding this correctly, then we might have to disable the expectation for Spark if it is qualitatively wrong without the use of `row_number()` (and incurring the corresponding performance penalty). Thank you!

Reply:
@alexsherstinsky I was able to make changes to `self_check/util.py` to address this and reverted this test back to its original state. Please let me know your thoughts on the code, but the short version is we are now sorting `unexpected_values` and `partial_unexpected_values` in both cases where `test["exact_match_out"]` is `True` or `False`.